_posts/2020-11-02-dynamically-changeable-logging.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: post
+title:  "Dynamically changeable logging levels"
+date: 2020-11-02
+author: stanislav_knot
+---
+
+Logging is an indisputably important feature for applications.
+Logging monitors what the application is doing, but equally importantly,  it logs what the application is not doing. 
+Looking into application logs can help us prevent an application remaining or even getting into a faulty state.
+
+<!--more-->
+
+
+## Setting logging levels dynamically
+
+There are many frameworks used for logging in Java applications, for example [`Logback`](http://logback.qos.ch/), [`tinylog`](https://tinylog.org/v2/), or [`log4j`](http://logging.apache.org/log4j/2.x/index.html).
+Because `log4j` is used in Kafka, we do use the same framework for Strimzi operator. 
+Strimzi components are split into the two groups regarding which implementation of `log4j` they use. 
+
+`log4j`
+- Kafka brokers
+- Kafka Connect
+- Kafka MirrorMaker2
+
+`log4j2`
+- Strimzi Kafka Bridge
+- Topic Operator
+- User Operator
+- Strimzi Cluster Operator
+
+`Log4j2` implementation supports dynamic reloading of logging configuration.
+`Log4j` supports dynamic changes but not reloading entire configuration file in runtime.
```

Comment:
Again ... I do not have a strong opinion about this. But if I didn't know what it probably meant from the implementation, I would probably not understand the difference. Should it maybe say more something like _log4j doesn't support reloading the whole log configuration from properties file, but it can be changed programatically using the Log4j APIs_?